src/txn/txn.c

```diff
@@ -904,7 +908,7 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 					    &(*updp)->timestamp,
 					    &txn->commit_timestamp);
 					(*updp)->prepare_state =
-					    WT_PREPARE_READY;
+					    WT_PREPARE_RESOLVED;
```

Comment:
This code doesn't have an associated `WT_WRITE_BARRIER()/WT_PUBLISH()` pair when setting `WT_PREPARE_LOCKED/WT_PREPARE_RESOLVED`. Is that correct?

Reply:
Yes, this code doesn't have an associated ` WT_WRITE_BARRIER()/WT_PUBLISH()`. As per my understanding `WT_WRITE_BARRIER()/WT_PUBLISH()` is not required as ref is in  `WT_REF_LOCKED` state.  Is my understanding correct?

Reply:
If that's the case, then there's no need to set the `prepare_state`, basically, it's one or the other, I think.